Source/Initialization/InitSpaceChargeField.cpp

```diff
@@ -0,0 +1,146 @@
+
+#include <AMReX_ParallelDescriptor.H>
+#include <AMReX_MLMG.H>
+#include <AMReX_MLNodeLaplacian.H>
+
+#include <WarpX.H>
+
+using namespace amrex;
+
+void
+WarpX::InitSpaceChargeField (WarpXParticleContainer& pc)
+{
+
+#ifdef WARPX_DIM_RZ
+    amrex::Abort("The initialization of space-charge field has not yet been implemented in RZ geometry.");
+#endif
+
+    // Allocate fields for charge and potential
+    const int num_levels = max_level + 1;
+    Vector<std::unique_ptr<MultiFab> > rho(num_levels);
+    Vector<std::unique_ptr<MultiFab> > phi(num_levels);
+    const int ng = WarpX::nox;
+    for (int lev = 0; lev <= max_level; lev++) {
+        BoxArray nba = boxArray(lev);
+        nba.surroundingNodes();
+        rho[lev].reset(new MultiFab(nba, dmap[lev], 1, ng)); // Make ng big enough/use rho from sim
+        phi[lev].reset(new MultiFab(nba, dmap[lev], 1, 0));
+        phi[lev]->setVal(0.);
+    }
+
+    // Deposit particle charge density (source of Poisson solver)
+    bool local = false;
+    bool reset = true;
```

Comment:
These are the only ones that I found, so I have to point them out:
could these two `bools above be `const bool`? :)